packages/graphql-hooks/src/GraphQLClient.js

```diff
@@ -90,14 +90,20 @@ class GraphQLClient {
   }
   /* eslint-enable no-console */
 
-  generateResult({ fetchError, httpError, graphQLErrors, data }) {
+  generateResult({
+    fetchError,
+    httpError,
+    graphQLErrors,
+    data,
+    responseProxy = {}
+  }) {
     const errorFound = !!(
       (graphQLErrors && graphQLErrors.length > 0) ||
       fetchError ||
       httpError
     )
     return !errorFound
-      ? { data }
+      ? { ...responseProxy, data }
```

Comment:
I think that would be better to append the result to a new key (e.g. `responseData`) instead of merging

It would be easier to add typescript types this way.

Reply:
makes sene 👍  adjusted that as wellw